decidim-debates/app/views/decidim/debates/debates/show.html.erb

```diff
@@ -41,18 +41,31 @@ edit_link(
     <% elsif admin_allowed_to?(:close, :debate, debate: debate) %>
       <%= link_to t(close_debate_action_text), Decidim::EngineRouter.admin_proxy(debate.component).edit_debate_debate_close_path(debate_id: debate.id, id: debate.id), class: "button hollow expanded button-sc button--icon follow-button" %>
     <% end %>
+
     <div class="card extra">
-      <div class="card__content">
+      <div class="card__content--multiple">
         <% if debate.start_time.present? && debate.end_time.present? %>
-          <div class="extra__date">
-            <%= l debate.start_time, format: "%d" %> <span class="extra__month"><%= l debate.start_time, format: "%B" %></span>
-          </div>
-          <div class="extra__time">
-            <%= l debate.start_time, format: "%H:%M" %> - <%= l debate.end_time, format: "%H:%M" %>
-          </div>
-        <% end %>
+  <div class="extra__date">
+    <% if debate.start_time.strftime("%d") == debate.end_time.strftime("%d") %>
+        <div>
+          <%= l debate.start_time, format: "%d" %> <span class="extra__month"> <%= l debate.start_time, format: "%B" %></span>
+          <div class="extra__time text-small"> <%= l debate.start_time, format: "%H:%M" %> - <%= l debate.end_time, format: "%H:%M" %></span></div>
+        </div>
+            <% else %>
+        <div>
+          <%= l debate.start_time, format: "%d" %> <span class="extra__month"> <%= l debate.start_time, format: "%B" %></span>
+          <div class="extra__time text-small"> <%= l debate.start_time, format: "%H:%M" %></span></div>
+        </div>
+        <%= icon "arrow-thin-right", class: "icon-arrow-thin-right" %>
+        <div>
+          <%= l debate.end_time, format: "%d" %> <span class="extra__month"> <%= l debate.end_time, format: "%B" %></span>
+          <div class="extra__time text-small"> <%= l debate.end_time, format: "%H:%M" %></span></div>
+        </div>
+            <% end %>
+    </div>
+            <% end %>
```

Comment:
Nice! It's working as it should :) To improve it, I would:

- Run `erblint` to correct the indentation (check the decidim docs if you're not sure how to do it)
- Extracting the logic into a [Rails helper](https://www.rubyguides.com/2020/01/rails-helpers/) would be super super nice 👌🏽  That way this file would be more readable and therefore easy to understand.